### PR TITLE
Incorporate screen-urls.php into website architecture

### PR DESCRIPTION
**Pull request by Sammidysam**
This is pretty straightforward:
- the old page `/community-voices/public/screen-urls.php` still exists, but now is generated through the website architecture
- there is a new page for accessing this which aligns with the standard URL conventions better, `/community-voices/locations`
- the new page uses the styling provided easily through the system architecture, as I figure why would we not want it?
- the new locations page links to the digital signage pages so that URLs don't need to be copied
- only logged-in admins can access the new locations page

Outstanding issues / things to work on next:
- how to give a more verbose error message in `App/Api/Component/SecureContainer.php` than "No decorated instance!"
- LocationLookup has a findAll2 because findAll does not fit the normal method of construction and fixing all of that was outside of the scope of this branch
- `Model/Component/StateObserver.php` should be given more helper methods to determine strings from classes so that objects can be passed around rather than strings

I would say each outstanding issue can be its own branch / discussion.

@jeremyfifty9 all look good to merge?  Is leaving the old screen-urls link okay for the purpose of backwards compatibility, or is it not a bother?  Which of the next items is best to work on next, and should they be made into Trello tasks?  Or only some?

### CODE REVIEW

**Comment by Sammidysam** [comment]
Update: the second two items in the outstanding issues list are already Trello tasks.  So it's really a question of just what your idea for the first one is.

File: src/CommunityVoices/App/Api/Component/SecureContainer.php [hunk 1]
```diff
@@ -24,15 +24,17 @@ public function contain($decoratedInstance)
             $user = $this->identifier->identify();
 
             if (!is_object($contained)) {
+                // This should be more verbose, but we cannot print out a class name
+                // of a non-object. So, what to do here?
                 throw new \Exception('No decorated instance!');
```
Comment (jeremyfifty9) [inline src/CommunityVoices/App/Api/Component/SecureContainer.php]: Maybe 'A secure container must have a container object' or something?

File: src/CommunityVoices/App/Api/Component/SecureContainer.php [hunk 2]
```diff
@@ -24,15 +24,17 @@ public function contain($decoratedInstance)
...
             }
 
+            $signature = get_class($contained) . "::" . $method;
+
             if (!method_exists($contained, $method)) {
-                throw new \Exception('Method not found');
+                throw new \Exception('Method not found ' . $signature);
```
Comment (jeremyfifty9) [inline src/CommunityVoices/App/Api/Component/SecureContainer.php]: Method '%signature%' not found I like better

File: src/CommunityVoices/Model/Mapper/LocationCollection.php [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+use CommunityVoices\Model\Entity;
+
+class LocationCollection extends DataMapper
+{
+    public function fetch(Entity\LocationCollection $container)
+    {
+        $this->fetchAll($container);
+    }
+
+    public function fetchAll(Entity\LocationCollection $container)
```
Comment (jeremyfifty9) [inline src/CommunityVoices/Model/Mapper/LocationCollection.php]: change naming -- these are collections (composites) `$locationCollection` is better.

File: src/CommunityVoices/Model/Mapper/LocationCollection.php [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    {
+        $query = "SELECT id, label, end_use FROM `community-voices_locations` ORDER BY end_use DESC, label ASC";
+        foreach ($this->conn->query($query) as $row) {
```
Comment (jeremyfifty9) [inline src/CommunityVoices/Model/Mapper/LocationCollection.php]: can you use `addEntityFromParams`? mappers do not need express knowledge of how to create domain objects

**Comment by Sammidysam** [comment]
Thanks for the feedback there!  Yeah I kind of just moved code into new locations without really thinking much about it.  I'll get on that tomorrow.

File: src/CommunityVoices/App/Api/Component/SecureContainer.php (hunk 2 above)
Comment (Sammidysam) [reply src/CommunityVoices/App/Api/Component/SecureContainer.php]: Hmm... I'm really used to error messages being in the format of the error message followed by the description of the specific case that helps. (e.g. Array index out of bounds: 4) Are you sure you prefer this?  It makes it harder to find what signature is causing the problem at a glance.

File: src/CommunityVoices/Model/Mapper/LocationCollection.php (hunk 3 above)
Comment (Sammidysam) [reply src/CommunityVoices/Model/Mapper/LocationCollection.php]: I figure https://github.com/EnvironmentalDashboard/community-voices/pull/19/commits/532e5b7e29d28ad98af83787c97ec8ffbeddfa8f covers what you mean?

File: src/CommunityVoices/Model/Mapper/LocationCollection.php (hunk 4 above)
Comment (Sammidysam) [reply src/CommunityVoices/Model/Mapper/LocationCollection.php]: Does https://github.com/EnvironmentalDashboard/community-voices/pull/19/commits/532e5b7e29d28ad98af83787c97ec8ffbeddfa8f cover this entirely?  Is it still fine to do the `$this->conn->query($query)` in the `foreach`?  The old code that I referenced had `prepare` called more often, so I don't know if this is good practice still.

**Comment by Sammidysam** [comment]
I'm also wondering what is best to work on after this pull request - I figure the findAll revisions across Services?  I'd prefer to work on the state observer first, but I figure that having two findAlls is kind of glaringly ugly.

File: src/CommunityVoices/App/Api/Component/SecureContainer.php (hunk 1 above)
Comment (jeremyfifty9) [reply src/CommunityVoices/App/Api/Component/SecureContainer.php]: i would not question that -- that sounds like a good idea.

File: src/CommunityVoices/Model/Mapper/LocationCollection.php (hunk 4 above)
Comment (jeremyfifty9) [reply src/CommunityVoices/Model/Mapper/LocationCollection.php]: did you test this? I did not, but I figure it won't work. look at how other SELECT queries use 'AS'. i figure it's OK to leave it in the for()

Comment (Sammidysam) [reply src/CommunityVoices/Model/Mapper/LocationCollection.php]: The query itself is the same one used historically - I copied that line from the old code.  So it both works here and also has worked for history.

File: src/CommunityVoices/App/Api/Component/SecureContainer.php (hunk 1 above)
Comment (Sammidysam) [reply src/CommunityVoices/App/Api/Component/SecureContainer.php]: Would that class file go in the same folder then?

Comment (Sammidysam) [reply src/CommunityVoices/App/Api/Component/SecureContainer.php]: Creating an empty class that extends `\Exception` accomplishes the goal I had in mind (see https://github.com/EnvironmentalDashboard/community-voices/commit/aa67f915dd384f67114aa8e753a90b172fb24f1b).  Now, it says uncaught `SecureContainerException` which I think is good.  Do you see any reason to extend its behavior any further at all?

File: src/CommunityVoices/Model/Mapper/LocationCollection.php (hunk 4 above)
Comment (Sammidysam) [reply src/CommunityVoices/Model/Mapper/LocationCollection.php]: `end_use` does actually line up correctly.  I did add code to make sure that ID is an integer, but all looks good here.

Comment (jeremyfifty9) [reply src/CommunityVoices/Model/Mapper/LocationCollection.php]: oh. i see. php method names are case insensitive.

**Pull request merged**